src/pal/src/locale/unicode.cpp

```diff
@@ -1632,3 +1641,68 @@ WideCharToMultiByte(
     PERF_EXIT(WideCharToMultiByte);
     return retval;
 }
+
+extern char g_szCoreCLRPath[MAX_PATH];
+
+/*++
+Function :
+
+PAL_BindResources - bind the resource domain to the path where the coreclr resides
+
+--*/
+VOID
+PALAPI
+PAL_BindResources(IN LPCSTR lpDomain)
+{
+#ifndef __APPLE__
+    char coreCLRDirectoryPath[MAX_PATH];
+
+    DWORD size = FILEGetDirectoryFromFullPathA(g_szCoreCLRPath, MAX_PATH, coreCLRDirectoryPath);
+    _ASSERTE(size <= MAX_PATH);
+    bindtextdomain(lpDomain, coreCLRDirectoryPath);
```

Comment:
Ah, ok. I have thought this function can not fail, but looking at the doc again, I can see that it actually can due to OOM. I'll add the propagation.